File: mono/nn/Edge.cs
using System;

namespace nn
{
	public class Edge <T> where T : FeatureRepresentation
	{
		private T a;
		private T b;

		public T A
		{
			get
			{
				return a;
			}
			set
			{
				a=value;
			}
		}
		public T B
		{
			get
			{
				return b;
			}
			set
			{
				b=value;
			}
		}

			public Edge(T a, T b):base()
			{
				this.a = a;
				this.b = b;
			}
			public bool contains(T node)
			{
				return ((a.Id==node.Id)||(b.Id==node.Id));
			}
			public T getTheOther(T node)
			{
				T rv=null;
				if(a.Id==node.Id)
					rv=b;
				else if(b.Id==node.Id)
					rv=a;
				return rv;
			}
	}
}

File: python/Edge.py
class Edge:
	'doc'
	
	def __init__(self,left,right):
		self.a=left
		self.b=right

	def contains(self,node):
		return ((self.a.getId()==node.getId())or(self.b.getId()==node.getId()))

	def getTheOther(self, node):
		if self.a.getId()==node.getId():
			rv=self.b
		elif self.b.getId()==node.getId():
			rv=self.a
		return rv

	def getA(self):
		return self.a;

	def getB(self):
		return self.b;


